simpleviewinc/keg-test-consumer
# Consumer Container

> Note: this PR is paired with [this PR](https://github.com/simpleviewinc/tap-events-force/pull/50) from `tap-events-force`. They have the same testing instructions, and either can be reviewed/merged first

# Context
* we need to be frequently testing the tap with `keg-test-consumer` to verify the build works as expected
* right now that's a laborious process

# Goal
* speed up testing 
* run everything in docker

# Updates
* added a `README.md` that provides testing instructions
* added container folder with injected app files
* updated `package.json` with some scripts for testing taps
* added `react-app-rewired` to configure webpack in `config-overrides.js`

### PR review updates
* simplified the `Dockerfile` so that we don't build from keg-base, which this does not need
* updated `run.sh` to start the consumer when the container starts

# Testing
* `keg evf && keg pr 50`
* `keg consumer && keg pr 1`

1. `keg consumer start`
  * starts the container itself, but not the react app yet, with the consumer mounted at `keg/app`
2. `keg consumer sync evf:install:start`
  * syncs your local tap-events-force into the container at `/keg/tap`, then starts the build server, which will rebuild the sessions component whenever a change is detected. It will automatically
    place it inside of the consumer's `/keg/app/node_modules/@keg-hub/tap-evf-sessions` 
3. `keg consumer att`
  * attach to the consumer container
4. `yarn start`
  * starts the react app
5. Verify you can see the sessions component at `local.kegdev.xyz`
6. Make a change in the tap (an easy one: change line `16` of `src/theme/containers/gridContainer` to a different color)
7. Verify you see it rebuilding the sessions component in the terminal output from step 2
8. Verify you see the consumer terminal output from step 4 recompiling the bundle
9. Verify the app automatically reloaded and displays your changes
